roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,698 @@
+                                               node['metadata']['name'])
+                                              for node in nodes]
+                          for addr in addrs
+                          if addr['type'] == 'InternalIP']
+            node_info = '\n'.join([' '.join(fields) for fields in node_addrs])
+            self.register_file('nodeinfo', node_info)
+
+            pod_addrs = [(pod['spec']['nodeName'],
+                          pod['status']['hostIP'],
+                          pod['metadata']['name'],
+                          pod['metadata']['namespace'],
+                          pod['status']['podIP'],
+                          pod['status']['containerStatuses'][0]['containerID']
+                          .replace('docker://', ''))
+                         for pod in pods
+                         if 'containerStatuses' in pod['status'] and
+                         pod['status']['containerStatuses'][0].get('ready') and
+                         'hostNetwork' not in pod['spec']]
+            pod_info = '\n'.join([' '.join(fields) for fields in pod_addrs])
+            self.register_file('podinfo', pod_info)
+
+            service_addrs = [(service['metadata']['name'],
+                              service['metadata']['namespace'],
+                              service['spec']['clusterIP'],
+                              service['spec']['ports'][0]['protocol'].lower(),
+                              str(service['spec']['ports'][0]['port']))
+                             for service in services
+                             if service['spec'].get('clusterIP') is not None]
+            service_info = '\n'.join([' '.join(fields)
+                                      for fields in service_addrs])
+            self.register_file('serviceinfo', service_info)
+
+        if not nodes:
+            self.register_failure(
+                'No nodes appear to be defined according to the API.'
+            )
+
+        for node in nodes:
+            self.check_node_kubelet(node)
+
+    def get_resource(self, kind):
+        """Return a list of all resources of the specified kind."""
+        for resource in self.task_vars['resources']['results']:
+            if resource['item'] == kind:
+                return resource['results']['results'][0]['items']
+
+        raise OpenShiftCheckException('CouldNotListResource',
+                                      'Could not list resource %s' % kind)
+
+    def check_node(self):
+        """Gather diagnostic information on a node and perform connectivity
+        checks on pods and services."""
+        node_name = self.get_node_name()
+        if not node_name:
+            self.register_failure(
+                'Could not find node name in node-config.yaml.'
+            )
+            return
+
+        if self.want_full_results:
+            try:
+                # XXX: Loading node logs can take in inordinate amount of time
```

Comment:
Same suggestions as above... but perhaps we should have a way to filter which nodes to grab logs from?  Often you just want the source and destination node logs.

Reply:
The set of hosts can be selected when running the playbook.  For example: `ansible-playbook -i hosts -l master-host-foo,node-host-bar,localhost openshift-ansible/playbooks/openshift-checks/adhoc.yml -e openshift_checks=sdn -e openshift_checks_output_dir=output` (`localhost` must always be specified because openshift-ansible always needs to do some local tasks).  Is that sufficient, or do you want an option specifically for selecting which nodes from which to read the journal?